src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2908,6 +2908,23 @@ public static void DCS_BasicRoundtripDCRVerifyWireformatScenarios()
         SerializationTestTypes.ComparisonHelper.CompareRecursively(value3, actual3);
     }
 
+    [Fact]
+    public static void DCS_ResolveNameVariationTest()
+    {
+        SerializationTestTypes.ObjectContainer instance = new SerializationTestTypes.ObjectContainer(new SerializationTestTypes.UserTypeContainer());
+        var setting = new DataContractSerializerSettings()
+        {
+            DataContractResolver = new SerializationTestTypes.UserTypeToPrimitiveTypeResolver(),
+            MaxItemsInObjectGraph = int.MaxValue,
+            IgnoreExtensionDataObject = false,
+            PreserveObjectReferences = false
```

Comment:
It seems that we only need to set `DataContractResolver`. Others are set to default value. Can you please remove the other settings so that the purpose of the test is more clear.